### PR TITLE
Event handler: introduce virtual desktop announcement switch handling

### LINKED ISSUE #5641: Windows 10: NVDA does not report which desktop I am on when I switch amongst multiple desktops
NVDA does not report which desktop I am on when I switch among multiple desktops in windows 10 Professional. To duplicate this, do the following. From with windows 10:
1. Get to the desktop with windows+d or to the task bar with windows+m.
2. Hit windows+ctrl+d to create a new desktop. 
Note:
You will not be told that a new desktop has been created. I believe sighted users do not get a notification either but there is reference to the "show desktop" button appearing.
3. Use control+windows+right arrow to switch to the next desktop.

You will not hear any output from NVDA. I would expect NVDA to say something like desktop1, desktop 2 etc.
Note:
You can hit ctrl+windows+left arrow to switch to a previous desktop but NVDA will not speak which desktop I am on.

As regards the use case for such a notification:
1. I may not remember what state I left that desktop in therefore it is handy to know that the desktop has changed.
2. I may want to skip that desktop and go to another desktop so the notification will help with rapid navigation.

I am using windows 10 professional 64bit and NVDA 2015.4.

### PR DESCRIPTION
**Pull request by josephsl**
Note: this is take 3.1 of #5641 to see if a PR build can be built:

### Link to issue number:
Fixes #5641 

### Summary of the issue:
In Windows 10 and later, NVDA does not announce virtual desktop names wen they are opened, switched, and closed.

### Description of user facing changes
NVDA will announce names of virtual desktops when opening, switching, and closing them.

### Description of development approach
Event handler is extended to include a virtual desktop switch handler and an extension to execute event to catch a name change event coming from CSRSS.

### Testing strategy:
Manual testing (prerequisites: Windows 10 or later):

1. Press Control+Windows+D to open virtual desktops
2. Switch between virtual desktops.
3. Close a virtual desktop.

### Known issues with pull request:
In some cases, closing a virtual desktop wil announce the focused control first and then announce virtual desktop name, but this is due to focus event firing before name change event.

### Change log entries:
New features:

In Windows 10 May 2019 Update and later, NVDA can announe virtual desktop names when opening, changing, and closing them. (#5641)

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by josephsl** [comment]
CC @SeanBudd, @JCSTeh, @michaelDCurran

File: source/eventHandler.py [hunk 1]
```diff
@@ -37,6 +38,12 @@
 #: the last object queued for a gainFocus event. Useful for code running outside NVDA's core queue 
 lastQueuedFocusObject=None
 
+
+# Handle virtual desktop switch announcements in Windows 10 and later
+virtualDesktopName: Optional[str] = None
+canAnnounceVirtualDesktopNames: bool = winVersion.getWinVer() >= winVersion.WIN10_1903
```
Comment (seanbudd) [inline source/eventHandler.py:44]: are these expected to be consumed by add-ons? perhaps these should be private

File: source/eventHandler.py [hunk 2]
```diff
@@ -287,19 +294,45 @@ def executeEvent(
...
 		elif not sleepMode:
-			_EventExecuter(eventName,obj,kwargs)
-	except:
-		log.exception("error executing event: %s on %s with extra args of %s"%(eventName,obj,kwargs))
+			_EventExecuter(eventName, obj, kwargs)
+	except Exception:
+		log.exception(f"error executing event: {eventName} on {obj} with extra args of {kwargs}")
+
+
+def handlePossibleDesktopNameChange() -> None:
```
Comment (seanbudd) [inline source/eventHandler.py:323]: same here - this should probably be private

**Review by seanbudd — COMMENTED** [review]
Thanks @josephsl - the general approach here looks good

**Comment by josephsl** [comment]
Hi, at least the function name (handle virtual desktop switch) is needed by Windows App Essentials add-on to determine NVDA Core patching determination. Others might become private, and I will take care of it in an upcoming commit. Thanks.

**Comment by seanbudd** [comment]
Hi @josephsl - there are some strange artifacts in the diff here from other commits, including CLDR changes

**Comment by josephsl** [comment]
Hi, when I did git pull origin master, it somehow did a rebase, and the CLDR commit artifacts came from git submodule update –init. Thanks.

**Comment by CyrilleB79** [comment]
> Hi, when I did git pull origin master, it somehow did a rebase, and the CLDR commit artifacts came from git submodule update –init. Thanks.

I do not know how to explain this. Maybe you have something strange in your git config when doing `git pull`.

Anyway, no only the CLDR submodule but also the symbol files appear in the modified files of this PR. This should be fixed please. Thanks.

**Comment by josephsl** [comment]
Hi, I see. I’ll do something about this. Thanks

**Comment by XLTechie** [comment]
I noticed the same thing happening on @josephsl commits a week or two back, but 
I think that PR was closed.

See the first answer here for what probably happened: 
https://stackoverflow.com/questions/9264314/git-commits-are-duplicated-in-the-same-branch-after-doing-a-rebase

TL;DR: you may have pulled because git told you to, but you should have force 
pushed.

**Comment by josephsl** [comment]
Should be resolved - did a force push as noted by a source from Luke's post above.

**Review by seanbudd — APPROVED** [review]
Thanks @josephsl
